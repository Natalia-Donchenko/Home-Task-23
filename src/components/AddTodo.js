import React, { useState } from 'react';
import AddBtn from './btns/AddBtn';

const AddTodo = () => {
  const [ state, setState ] = useState({
    text: ''
  });
  const { text } = state;
   
  const handleChange = (e) =>{
    setState({...state, [e.target.name]: e.target.value});
  };

  return  (
    <>
    <div className='addTask'>
      <input 
        className="task-input"
        placeholer='What do you want to do?'
        type='text' 
        value={text} 
        name='text' 
        onChange={handleChange} 
      />

      <AddBtn 
        setState={setState}
        state={state} 
        text = {text}
      />
    </div>
    </>
  );
};

export default AddTodo;
