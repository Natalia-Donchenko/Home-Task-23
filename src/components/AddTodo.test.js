import configureStore from 'redux-mock-store'
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import AddTodo from './AddTodo';

const mockStore = configureStore([]);

describe("My connected React-Redux component AddTodo", () => {
  let store;
  let component;

  beforeEach(() => {
    store= mockStore({
      myState: "sapmle text"
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <AddTodo />
      </Provider>
    );
  });

  it("Should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);

  });
})