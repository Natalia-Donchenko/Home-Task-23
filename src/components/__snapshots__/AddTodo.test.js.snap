// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My connected React-Redux component AddTodo Should render with given state from Redux store 1`] = `
<div
  className="addTask"
>
  <input
    className="task-input"
    name="text"
    onChange={[Function]}
    placeholer="What do you want to do?"
    type="text"
    value=""
  />
  <button
    className="addTask-btn"
    onClick={[Function]}
    type="button"
  >
    add
  </button>
</div>
`;
